lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1210 @@
+        'fu_number_sick'
+    ]
+
+    string_questions = [
+        'fu_fever',
+        'fu_headache',
+        'fu_cough',
+        'fu_chills',
+        'fu_sweat',
+        'fu_throat',
+        'fu_nausea',
+        'fu_nose',
+        'fu_tired',
+        'fu_ache',
+        'fu_breathe',
+        'fu_diarrhea',
+        'fu_rash',
+        'fu_ear',
+        'fu_eye',
+        'fu_smell_taste',
+        'fu_feel_normal',
+        'fu_care',
+        'fu_hospital_ed',
+        'fu_work_school',
+        'fu_activities',
+        'fu_which_activites',
+        'fu_test_result',
+        'fu_behaviors_no',
+        'fu_behaviors_inconclusive',
+        'fu_behaviors',
+        'fu_1_symptoms',
+        'fu_2_symptoms',
+        'fu_3_symptoms',
+        'fu_4_symptoms',
+        'fu_1_test',
+        'fu_2_test',
+        'fu_3_test',
+        'fu_4_test',
+        'fu_1_result',
+        'fu_2_result',
+        'fu_3_result',
+        'fu_4_result',
+        'fu_healthy_test',
+        'fu_healthy_result'
+    ]
+
+    date_questions = [
+        'fu_timestamp',
+        'fu_symptom_duration',
+        'fu_date_care',
+        'fu_1_date',
+        'fu_2_date',
+        'fu_3_date',
+        'fu_4_date',
+        'followup_date'
+    ]
+
+    question_categories = {
+        'valueBoolean': boolean_questions,
+        'valueInteger': integer_questions,
+        'valueString': string_questions,
+        'valueDate': date_questions
+    }
+
+    # Combine checkbox answers into one list
+    checkbox_fields = [
+        'fu_care',
+        'fu_which_activites',
+        'fu_behaviors_no',
+        'fu_behaviors_inconclusive',
+        'fu_behaviors',
+        'fu_1_symptoms',
+        'fu_2_symptoms',
+        'fu_3_symptoms',
+        'fu_4_symptoms'
+    ]
+
+    for field in checkbox_fields:
+        record[field] = combine_checkbox_answers(record, field)
+
+    return questionnaire_response(record, question_categories, patient_reference, encounter_reference)
+
+def create_testing_determination_internal_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the
+    testing_determination_internal instrument.
+    """
+
+    string_questions = [
+        'testing_trigger',
+        'testing_type'
+    ]
+
+    question_categories = {
+        'valueString': string_questions
+    }
+
+    return questionnaire_response(record, question_categories, patient_reference, encounter_reference, True)
+
+def create_computed_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict,
+                                            birthdate: datetime, encounter_date: datetime) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for a
+    computed questionnaire.
+    """
```

Comment:
it's not clear to me why this QuestionnaireResponse needs to created separately from other QuestionnaireResources, and furthermore, why this field needs to be calculated at all. Age should exist already as a calculated field .

Reply:
I separated the questionnaires conceptually.
Age in REDCap (core_age_years) is the participant's age when s/he enrolled. That may not be the age at the time of the encounter.

Reply:
Thanks, could you please add more information to this function's docstring to elucidate its purpose?

Reply:
I've added this to the docstring:
This "computed questionnaire" produces
    answers that do not appear directly in an actual REDCap survey.
    For example, a computed question captures the participant's age
    on the date of the encounter.